### ISSUE TITLE
Can't install with Phoenix 1.3 and Elixir 1.4

### ISSUE THREAD

**Issue by abitdodgy** [geolessel/react-phoenix#23]
I'm trying the 0.4.3 release. I keep getting this error:

```
** (Mix) Could not start application react_phoenix: ReactPhoenix.Application.start(:normal, []) returned an error: shutdown: failed to start child: ReactPhoenix.ReactIo
    ** (EXIT) shutdown: failed to start child: ReactPhoenix.ReactIo.Pool
        ** (EXIT) an exception was raised:
            ** (MatchError) no match of right hand side value: {:error, {:function_clause, [{Porcelain, :spawn_shell, [nil, [in: :receive, out: {:send, #PID<0.541.0>}]], [file: 'lib/porcelain.ex', line: 241]}, {StdJsonIo.Worker, :init, 1, [file: 'lib/std_json_io/worker.ex', line: 12]}, {:gen_server, :init_it, 6, [file: 'gen_server.erl', line: 328]}, {:proc_lib, :init_p_do_apply, 3, [file: 'proc_lib.erl', line: 247]}]}}
                /Users/mohamad/Code/app/deps/poolboy/src/poolboy.erl:275: :poolboy.new_worker/1
                /Users/mohamad/Code/app/deps/poolboy/src/poolboy.erl:296: :poolboy.prepopulate/3
                /Users/mohamad/Code/app/deps/poolboy/src/poolboy.erl:145: :poolboy.init/3
                (stdlib) gen_server.erl:328: :gen_server.init_it/6
                (stdlib) proc_lib.erl:247: :proc_lib.init_p_do_apply/3
```

Any ideas?

**Comment by abitdodgy** [comment]
I got this to work by explicitly adding `react-stdio` to my `package.json` file. However, I'm getting the same error on Heroku when I deploy. Any ideas?

**Comment by abitdodgy** [comment]
I've been fighting this a couple of days, still no luck. This is one error message that I get consistently.

I even made a fresh install of OS X Sierra, wiped out the entire drive, and installed a fresh version of brew. I used it to install npm. After I that I installed my deps and went ran npm install. I seem to be able to get rid of it if I run `npm install` before `mix deps.get`
```
npm WARN assets No description
npm WARN The package react-stdio is included as both a dev and production dependency.
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: abbrev@1.1.0 (node_modules/react-phoenix/node_modules/fsevents/node_modules/abbrev):
npm WARN enoent SKIPPING OPTIONAL DEPENDENCY: ENOENT: no such file or directory, rename '/Users/mohamad/Code/app/deps/react_phoenix/node_modules/fsevents/node_modules/abbrev' -> '/Users/mohamad/Code/app/deps/react_phoenix/node_modules/fsevents/node_modules/.abbrev.DELETE'

npm ERR! path /Users/mohamad/Code/app/deps/react_phoenix/node_modules/fsevents/node_modules/ansi-regex
npm ERR! code ENOENT
npm ERR! errno -2
npm ERR! syscall rename
npm ERR! enoent ENOENT: no such file or directory, rename '/Users/mohamad/Code/app/deps/react_phoenix/node_modules/fsevents/node_modules/ansi-regex' -> '/Users/mohamad/Code/app/deps/react_phoenix/node_modules/fsevents/node_modules/.ansi-regex.DELETE'
npm ERR! enoent This is related to npm not being able to find a file.
npm ERR! enoent 

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/mohamad/.npm/_logs/2017-08-19T22_16_35_673Z-debug.log
Mohamads-MacBook-Pro:assets mohamad$ 
```

**Comment by abitdodgy** [comment]
So I'm a little closer to figuring this out. The way I managed to get the app to work locally was by following these steps:

1. Clean all dependencies and delete `package-lock.json` and `node_modules` folder from `assets`
2. Install app *without* `react-phoenix`.
3. cd into `assets` and execute `npm install`
4. Add `react-phoenix` and run `mix deps.get`
5. Add the react-phoenix js file to `package.json` "file:../deps/react_phoenix" then cd into `assets` and run `npm install`

This sequence manage to fix the errors on my local machine. It seems that, for some reason, it's necessary to run `npm install` before adding the react-phoenix to the app. Still can't get it to work on Heroku, though.

**Comment by hipertracker** [comment]
It does not work 
```
=INFO REPORT==== 2-Sep-2017::21:42:53 ===
    application: logger
    exited: stopped
    type: temporary
** (Mix) Could not start application react_phoenix: ReactPhoenix.Application.start(:normal, []) returned an error: shutdown: failed to start child: ReactPhoenix.ReactIo
    ** (EXIT) shutdown: failed to start child: ReactPhoenix.ReactIo.Pool
        ** (EXIT) an exception was raised:
            ** (MatchError) no match of right hand side value: {:error, {:function_clause, [{Porcelain, :spawn_shell, [nil, [in: :receive, out: {:send, #PID<0.271.0>}]], [file: 'lib/porcelain.ex', line: 241]}, {StdJsonIo.Worker, :init, 1, [file: 'lib/std_json_io/worker.ex', line: 12]}, {:gen_server, :init_it, 2, [file: 'gen_server.erl', line: 365]}, {:gen_server, :init_it, 6, [file: 'gen_server.erl', line: 333]}, {:proc_lib, :init_p_do_apply, 3, [file: 'proc_lib.erl', line: 247]}]}}
```

**Comment by sickill** [comment]
I'm having the same issue now. It used to work with Node 6 and NPM 3, but now after I upgraded to Node 8 and NPM 5 I get the above problem too.

**Comment by sickill** [comment]
@abitdodgy as a workaround I created a symlink at `assets/deps/react_phoenix` pointing to `../../deps/react_phoenix`, then changed `"file:../deps/react_phoenix"` to `"file:deps/react_phoenix"`.

**Comment by geolessel** [comment]
@sickill @abitdodgy I'm sorry you were having some issues there. I've just published v0.5.0 in which I've removed all the server-side rendering code. It apparently really only worked for me in the end because of some complex hand-wavy stuff I had done during the creation of the server-side code and I'm about 50/50 on replicating it even myself.

Almost every issue I see open here comes down to code that server-side rendering relies on. Please update to v0.5.0 to see if you continue to have issues.
